src/app/plugins/pubsub.plugin.ts

```diff
@@ -1,23 +1,138 @@
+    const subIn = this._parseInput(args || []);
+
+    // return the og message
+    if (!subIn) {
+      const state = this.container.storeService.get('pubSubStore').state;
+      if (state?.onSale) {
+        message.reply('Chicken tender subs are on sale :eyes:');
+      } else {
+        message.reply('Chicken tender subs are **NOT** on sale.');
+      }
+      return;
+    }
+
+    if (subIn === 'list' || subIn === 'types') {
+      //Simply return the list of supported subs
+      this._generateEmbedList();
+      message.reply(this._EMBED_LIST);
+      return;
+    }
+
+    // checks if their sub was a sub, then if that sub is recognised
+    const subType = this._SUBS.find((sub: string) => sub === subIn) || 'random';
+
+    //recieves the according info and posts, or derps
+    await this.container.httpService
+      .get(`${this._API_URL}/subs/?name=${subType}`)
+      .then((response: IHttpResponse) => {
+        const [subData] = response.data;
+        const embed = new MessageEmbed();
+
+        const parsedNamed = this._normaliseName(subData.sub_name);
+        const saleInfo =
+          subData.status.toLowerCase() === 'true' //a function would be too shallow to warrant making
+            ? `for the discounted price of ${subData.price} until ${
+                subData.last_sale.split('-')[1]
+              }`
+            : `it isn't on sale currently, but who could say no to ${subData.price}.  The last time it was on sale was ${subData.last_sale}`;
+
+        embed.setColor('#0099ff').setTitle(parsedNamed);
+
+        embed.setDescription(
+          `Get your own *${parsedNamed}* sub, ${saleInfo}. So what are you waiting for?  Come on down to Publix now to get yourself a beautiful sub. Just look at this beauty right here! 😍😍😍:pubsub::pubsub::pubsub:🤤🤤🤤` +
+            `\n\n *pulled from https://pubsub-api.dev.*`
+        );
+
+        embed.setImage(subData.image);
+
+        message.reply(embed);
+      })
+      .catch((err) => this.container.loggerService.warn(err));
+  }
+
+  private async _generateEmbedList(): Promise<void> {
+    const lastListUpdate: number = Date.now() - this._LAST_UPD_TIME;
+    if (lastListUpdate < this._SUB_UPD_THRESH) {
+      return;
+    }
+
+    this._LAST_UPD_TIME = Date.now(); //since it only updates once a day, don't need to worry about accuracy
+
+    this._updateList();
+
+    const numCols = Math.min(3, Math.ceil(this._SUBS.length / 10));
+    const numRows = Math.ceil(this._SUBS.length / numCols);
+
+    this._EMBED_LIST = new MessageEmbed();
+
+    this._EMBED_LIST.setColor('#0099ff').setTitle('Types of Subs');
+    const subs = this._SUBS.map((sub) => this._normaliseName(sub));
+
+    for (let cols = 0; cols < numCols; cols++) {
+      const columnSubs = subs.slice(numRows * cols, numRows * (cols + 1));
+
+      this._EMBED_LIST.addField(
+        `${columnSubs[0].charAt(0)} - ${columnSubs[columnSubs.length - 1].charAt(0)}`,
+        columnSubs.join('\n'),
+        true
+      );
```

Comment:
what does this do

Reply:
Creates a dynamic table based upon the length of the array.  You can look at the cats breed list to see what it ends up like when the array is big.

Reply:
Just to note I plan on making this a message service function in the future so it is only in one place.

Reply:
put a comment pls if we have to ask what it does lol